### PR TITLE
Vault Infra

### PR DESCRIPTION
**Pull request by DaMandal0rian**
This PR creates a hashicorp vault single cluster instance on AWS using terraform.
 - Backend storage S3 (encrypted / private acl)
 - KMS key to manage vault seal
 - IAM user to launch the ec2 instance with S3 access enabled.

This is required for the infrastructure to automate the process of handling keys and network deployments with CI and gitops.

Related Issue #131

### CODE REVIEW

File: vault/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  vpc_security_group_ids = [aws_security_group.vault_sg.id]
+
+  tags = {
+    Name = "vault-server"
+  }
+
+  provisioner "remote-exec" {
+  inline = [
+    "export DEBIAN_FRONTEND=noninteractive",
+    "sudo apt update -y",
```
Comment (nazar-pc) [inline vault/terraform/main.tf]: Just wondering, do all of these actually need sudo? I'd expect provisioner to run under root already

File: vault/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    "sudo apt upgrade -y",
+    "sudo apt install git curl wget gnupg openssl make build-essential jq net-tools unzip nginx certbot python3-certbot-nginx -y",
```
Comment (nazar-pc) [inline vault/terraform/main.tf]: Are all of these actually necessary? I don't see `make`, `jq` and others used anywhere.

File: explorer/.gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.terraform*
+*.tfstate*
+*.tfplan
+.DS_Store
+backend_config.hcl
+*.tfvars
```
Comment (nazar-pc) [inline explorer/.gitignore]: This seems to be committed by accident

File: vault/terraform/main.tf (hunk 1 above)
Comment (DaMandal0rian) [reply vault/terraform/main.tf]: yes, sudo is needed for ubuntu user.

File: explorer/.gitignore (hunk 3 above)
Comment (DaMandal0rian) [reply explorer/.gitignore]: good catch thanks.

File: vault/terraform/main.tf (hunk 2 above)
Comment (DaMandal0rian) [reply vault/terraform/main.tf]: I could remove some of them and install later if needed.

**Review by DaMandal0rian — COMMENTED** [review]
will make the minor changes

**Review by DaMandal0rian — COMMENTED** [review]
will make the minor changes

File: vault/terraform/main.tf (hunk 2 above)
Comment (nazar-pc) [reply vault/terraform/main.tf]: We should only install packages that are required for a specific machine, not the same set of packages on all machines regardless of the role. Latest version of this PR still installs with `wget` ans `curl`. Do we really actually need both though?

**Review by vedhavyas — COMMENTED** [review]
Looks good overall

File: vault/terraform/main.tf (hunk 2 above)
Comment (DaMandal0rian) [reply vault/terraform/main.tf]: done! removed wget.

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**